.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,50 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ main ]
+  # Require protected branches to no introduce security alerts
+  pull_request:
+    branches: [ main ]
+  # Keep results up to date if no activity happens
+  schedule:
+    - cron: '37 22 * * 1'
+  workflow_dispatch:
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'cpp' ]
```

Comment:
I wonder how C would work if it's optimized for C++. May be fine since C++ is a superset of C, but could produce interesting results.

Reply:
In CodeQL, C and C++ are treated the same so this is the correct setting for CodeQL